### PR TITLE
Add Suez Water config flow

### PR DESCRIPTION
**Pull request by joostlek**
## Proposed change

Add Suez Water config flow

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/104730
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/suez_water.markdown [hunk 1]
```diff
@@ -24,32 +25,6 @@ It also gets the following data:
...
-  required: true
-  type: string
-counter_id:
-  description: The Tout Sur Mon Eau counter id. 
-  required: true
-  type: integer
-{% endconfiguration %}
+{% include integrations/config_flow.md %}
 
 `counter_id` is the water counter id. It can be found on your _Tout Sur Mon Eau_ [user account](https://www.toutsurmoneau.fr/mon-compte-en-ligne/historique-de-consommation-tr). Look in the source code of the page for something similar to `url: '/mon-compte-en-ligne/statMData' + '/123456789'`. The `counter_id` in this case is `123456789`.
```
Comment (gjohansson-ST) [inline source/_integrations/suez_water.markdown:30]: No longer needed?

**Review by gjohansson-ST — DISMISSED** [review]

File: source/_integrations/suez_water.markdown (hunk 1 above)
Comment (joostlek) [reply source/_integrations/suez_water.markdown:30]: Oh right, that's right

Comment (joostlek) [reply source/_integrations/suez_water.markdown:30]: No wait, you still need it for suez

Comment (gjohansson-ST) [reply source/_integrations/suez_water.markdown:30]: Sorry, I confused it with the other one 👍

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @joostlek and @gjohansson-ST :+1:

**Pull request merged**